Replace `RMARKDOWNNOTEBOOK` with `QUARTONOTEBOOK`

delfiterradas opened the pull request:
Closes: https://github.com/nf-core/differentialabundance/issues/485
Closes: https://github.com/nf-core/differentialabundance/issues/486
Closes: https://github.com/nf-core/differentialabundance/issues/88

<!--
# nf-core/differentialabundance pull request

Many thanks for contributing to nf-core/differentialabundance!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

delfiterradas committed:
Install `quartonotebook` module from nf-core

delfiterradas committed:
Use quartonotebook in pipeline

antoniasaracco committed:
Fix: convert NA values to False in sample metadata table

grst committed:
Merge pull request #489 from antoniasaracco/issue_478

Fix: convert NA values to False in sample metadata table

delfiterradas committed:
Make quarto work with Rmd file

delfiterradas committed:
Modify module to take more than one template file

delfiterradas committed:
Update to run quarto for each template without modifying the nf-core module

atrigila committed:
feat: add dropdown menu in pca

atrigila committed:
Merge pull request #500 from atrigila/dropdown

delfiterradas committed:
Fix rendering

delfiterradas committed:
Merge branch 'dev_reports' into quarto

delfiterradas committed:
Fix errors

delfiterradas committed:
Update snapshots

delfiterradas committed:
Allow multiple report files

delfiterradas committed:
Add prefix to get as many reports as inputs for quarto and get single `.zip`

delfiterradas committed:
Define input_dir parameter in notebook

delfiterradas committed:
Add test with multiple inputs for quarto

delfiterradas committed:
Remove rmarkdown and minor changes

delfiterradas committed:
Remove unnecesary params from quarto

delfiterradas committed:
Select needed params to avoid errors if running with extra params

delfiterradas committed:
Create bundle with full meta

delfiterradas committed:
Fix bundle and update tests

delfiterradas committed:
Allow python templates for quarto

delfiterradas committed:
Modify documentation to reflect changes for Quarto

delfiterradas committed:
Update `.md` files

delfiterradas committed:
Add conda and singularity for Quarto notebook

delfiterradas committed:
Fix alignment

delfiterradas committed:
Update docs/usage.md

Co-authored-by: Alexander Peltzer <apeltzer@users.noreply.github.com>

delfiterradas committed:
Update nextflow_schema.json

delfiterradas committed:
Update workflows/differentialabundance.nf

Co-authored-by: Gregor Sturm <mail@gregor-sturm.de>

delfiterradas committed:
Remove unused section for `.qmd`

delfiterradas committed:
Switch to nested parameters in quarto

delfiterradas committed:
Fix: Add back quarto template file name to report name

delfiterradas commented on nextflow.config:
```diff
@@ -204,7 +199,7 @@ params {
     css_file                = "$projectDir/assets/nf-core_style.css"
     citations_file          = "$projectDir/CITATIONS.md"
-    report_file             = "$projectDir/assets/differentialabundance_report.Rmd"
+    report_file             = "$projectDir/assets/differentialabundance_report.qmd"
```
With @atrigila and @grst we were discussing if maybe this parameter should be renamed to report_files since now multiple template files can be used as input https://github.com/nf-core/differentialabundance/pull/499#discussion_r2375196365

delfiterradas committed:
Fix linting errors

delfiterradas committed:
Merge branch 'dev' into quarto

delfiterradas committed:
Fix report with decoupler and update tests

delfiterradas committed:
Revert linting updates

delfiterradas committed:
Fix path to report files in default test

atrigila commented on tests/quarto.qmd:
To give some context to reviewers:
- This is a dummy file that is used to test that the pipeline can now handle several template files as input. It is used in nf-tests. 
- In a future PR, this will be replaced by a QC/Results quarto report, respectively. (https://github.com/nf-core/differentialabundance/issues/369)

pinin4fjords replied:
If this is specifically for testing, I'd move it into the appropriate test dir so as not to confuse users

pinin4fjords commented:
Would it be possible for you to post the HTML-rendered notebooks before and after this?

pinin4fjords commented on workflows/differentialabundance.nf:
```diff
@@ -804,35 +804,52 @@ workflow DIFFERENTIALABUNDANCE {
+            .map {
+                def notebooks = it[0..-3]       // All elements except the last two (meta and input_files)
+                def meta = it[-2]               // Second to last element is meta
+                def input_files = it[-1]        // Last element is input files
```
Copilot's suggestion is wrong here obviously, but it's correct that the magic indices are unhelpful. Could you use the more specific syntax instead?

```
.map{ foo, bar ... ->
```

delfiterradas committed:
Minor style and bug fixes

delfiterradas committed:
Remove meta form param names

delfiterradas commented:
@pinin4fjords Yes, these are the before ([SRP254919_before.html](https://github.com/user-attachments/files/22626719/SRP254919_before.html)) and after ([SRP254919_after.html](https://github.com/user-attachments/files/22626791/SRP254919_after.html)) of the reports and a side by side comparison of first part:

<img width="1857" height="953" alt="image" src="https://github.com/user-attachments/assets/f4ffa6c4-42c4-455c-a0fb-bc018d5bdf51" />

grst commented on assets/differentialabundance_report.qmd:
```diff
@@ -492,19 +404,21 @@ names(differential_tables) <- names(sig_differential)
 
 
-A summary of `r params$observations_type` metadata is below:
+A summary of `{r} params$meta$params$observations_type` metadata is below:
```
@Copilot, this is incorrect. The inline syntax has been changed to `{r}` in quarto, see https://quarto.org/docs/computations/inline-code.html

pinin4fjords reviewed (commented):
Sorry about copilot's feedback, sometimes it's useful, sometimes it's not!

Just checked the output files (thank you for providing them and noticed a few things):

- The controls on colouring in the PCA are nice, but placed in a way that makes things look messy. Is it possible to move them? See the attached image.
- Points are no longer labelled in the 3D PCA. Is that intentional?
- Why the re-ordering of the contrasts? It should match the original contrast file (was the original wrong?)
- What happened to the Volcano plots in the differential section?


<img width="512" height="221" alt="Screenshot 2025-10-01 at 15 16 50" src="https://github.com/user-attachments/assets/2805d92a-4287-45bf-9ef2-e15d55aded77" />

delfiterradas committed:
Move `quarto.qmd` to tests folder

delfiterradas committed:
Sort the contrasts table

delfiterradas committed:
Print volcano plots

delfiterradas committed:
Fix position for drop down menu on PCA

delfiterradas committed:
Modify sintax for creation of bundle input

delfiterradas commented:
@pinin4fjords 
> * The controls on colouring in the PCA are nice, but placed in a way that makes things look messy. Is it possible to move them? See the attached image.

Yes, I have moved the position of the drop down menu. (see image below)
> * Points are no longer labelled in the 3D PCA. Is that intentional?

They are no longer labelled, but when you hover over the points the labels should be visible. This way if the plot has lots of points it shouldn't be so cluttered.

<img width="922" height="603" alt="image" src="https://github.com/user-attachments/assets/c3faccf7-1b03-4447-bed2-c157982adf7d" />

> * Why the re-ordering of the contrasts? It should match the original contrast file (was the original wrong?)

I think this depended on the order in which each contrast was emitted. I have now added a sorting so that the order is always preserved.
> * What happened to the Volcano plots in the differential section?

That was an error in the rendering, but they should now appear as before :)

<img width="973" height="846" alt="image" src="https://github.com/user-attachments/assets/5dd04d58-985c-4f69-85e3-a0c38a91a973" />

grst commented:
> Points are no longer labelled in the 3D PCA. Is that intentional?

Different topic, but is a 3D PCA on a 2D screen useful for anything?

pinin4fjords commented:
> > Points are no longer labelled in the 3D PCA. Is that intentional?
> 
> Different topic, but is a 3D PCA on a 2D screen useful for anything?

I find them useful, but only because you can move them around :-)

pinin4fjords commented:
Can I see a new copy of the improved report?

delfiterradas commented:
@pinin4fjords here is the most recent version of the report:
[SRP254919_adjusted.html](https://github.com/user-attachments/files/22722376/SRP254919_adjusted.html)

pinin4fjords reviewed (approved):
Looks fixed now, thanks!
